### PR TITLE
Simplify access to merge/rebase param for the given branch

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/CheckOutBranchAction.java [hunk 1]
```diff
@@ -30,7 +31,7 @@
  *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
  * </ul>
  */
-public class CheckOutBranchAction extends AnAction {
+public class CheckOutBranchAction extends AnAction implements DumbAware {
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/CheckOutBranchAction.java:34]: This action wasn't `DumbAware` so far... was there any reason or it's just been overlooked?

File: backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java [hunk 2]
```diff
@@ -41,30 +37,4 @@
   public Option<BaseGitMacheteBranch> getBranchByName(String branchName) {
     return branchByName.get(branchName);
   }
-
-  @Override
-  public IGitRebaseParameters getParametersForRebaseOntoParent(BaseGitMacheteNonRootBranch branch)
```
Comment (PawelLipski) [inline backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java:46]: We didn't need the repository for anything even here in the removed impl, this method could've as well been static

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/CheckOutBranchAction.java (hunk 1 above)
Comment (mkondratek) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/CheckOutBranchAction.java:34]: Nope, it's `dumbAware` :+1:

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/BaseRebaseBranchOntoParentAction.java [hunk 3]
```diff
@@ -90,23 +88,19 @@ public void update(AnActionEvent anActionEvent) {
 
   protected void doRebase(AnActionEvent anActionEvent, BaseGitMacheteNonRootBranch branchToRebase) {
     Project project = ActionUtils.getProject(anActionEvent);
-    Option<IGitMacheteRepository> gitMacheteRepository = getGitMacheteRepository(anActionEvent);
```
Comment (micpiotrowski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/BaseRebaseBranchOntoParentAction.java:93]: If I'm right `DataKeys#KEY_GIT_MACHETE_REPOSITORY` from javadoc on top of file can be removed

Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/BaseRebaseBranchOntoParentAction.java:93]: Done on another PR

**Pull request merged**